app/src/main/java/com/example/grosharies/ui/navigation/Navigator.kt

```diff
@@ -0,0 +1,145 @@
+package com.example.grosharies.ui.navigation
+
+import android.app.Application
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.navigation
+import com.example.grosharies.presentation.groceryList.GroceryListViewModel
+import com.example.grosharies.presentation.group.GroupViewModel
+import com.example.grosharies.presentation.listItem.ListItemViewModel
+import com.example.grosharies.presentation.nameInput.NameInputViewModel
+import com.example.grosharies.ui.Home
+import com.example.grosharies.ui.groceryList.EditList
+import com.example.grosharies.ui.groceryList.ListOverview
+import com.example.grosharies.ui.groceryList.NewList
+import com.example.grosharies.ui.groceryList.StartShopping
+import com.example.grosharies.ui.groups.*
+import com.example.grosharies.ui.listItem.ListDetail
+import com.example.grosharies.ui.listItem.ListItemNew
+
+@ExperimentalAnimationApi
+@Composable
+fun Navigator(navController: NavHostController) {
+    val context = LocalContext.current
+    val groupViewModel: GroupViewModel = viewModel(
+        factory = GroupViewModel.GroupViewModelFactory(context.applicationContext as Application)
+    )
+    val nameInputViewModel: NameInputViewModel = viewModel(
+        factory = NameInputViewModel.NameInputViewModelFactory(context.applicationContext as Application)
+    )
+    val listItemViewModel: ListItemViewModel = viewModel(
+        factory = ListItemViewModel.ListItemViewModelFactory(context.applicationContext as Application)
+    )
+    val groceryListViewModel: GroceryListViewModel = viewModel(
+        factory = GroceryListViewModel.GroceryListViewModelFactory(context.applicationContext as Application)
+    )
+
+    NavHost(
+        navController = navController,
+        startDestination = Screen.Home.route,
+        route = "root"
+    ) {
+        composable(route = Screen.Home.route) { Home(navController = navController) }
+        groupNavigation(navController, groupViewModel, listItemViewModel, nameInputViewModel)
+        listNavigation(navController, groceryListViewModel, listItemViewModel, nameInputViewModel)
+    }
+}
+
+@ExperimentalAnimationApi
+fun NavGraphBuilder.groupNavigation(
```

Comment:
How the graph is built up I like a little less, I'd handle navigation/screens more diff.

Roughly

`enum class Screen(val route: String, val title: String, val icon: ImageVector) {
    Style("style", "style", Icons.Filled.Home),
    CollapsingHeader("header", "header", Icons.Filled.Email),
    Animations("animations", "Anim", Icons.Filled.Star),
    Fourth("fourth", "fourth", Icons.Filled.Call),
}

@Preview(showBackground = true)
@Composable
fun MainScreenPreview() {
    ComposeTestApplicationTheme {
        MainScreen()
    }
}

@Composable
fun MainScreen() {
    val navController = rememberNavController()
    Scaffold(
        bottomBar = {
            BottomNavigation {
                val navBackStackEntry by navController.currentBackStackEntryAsState()
                val currentRoute = navBackStackEntry?.arguments?.getString(KEY_ROUTE)
                Screen.values().forEach { screen ->
                    BottomNavigationItem(
                        icon = { Icon(screen.icon, "") },
                        label = { Text(screen.title) },
                        selected = currentRoute == screen.route,
                        onClick = {
                            navController.navigate(screen.route) {
                                popUpTo = navController.graph.startDestination
                                launchSingleTop = true
                            }
                        }
                    )
                }
            }
        }
    ) {
        NavHost(
            navController,
            startDestination = Screen.Style.route,
            modifier = Modifier.padding(bottom = 56.dp),
        ) {
            for (screen in Screen.values()) {
                composable(screen.route) { ShowScreen(screen = screen) }
            }
        }
    }
}

@Composable
fun ShowScreen(screen: Screen) {
    when (screen) {
        Screen.Style -> StyleScreen()
        Screen.CollapsingHeader -> CollapsingHeaderScreen()
        Screen.Fourth -> ThedoScreen(viewModel = viewModel(ThedoViewModel::class.java))
        Screen.Animations -> AnimationsScreen()
    }
}`

Just don't understand why you create the viewModels here and re-create them in the screen.